daemon/logger/fluentd/fluentd.go

```diff
@@ -163,7 +163,7 @@ func (f *fluentd) Log(msg *logger.Message) error {
 	for k, v := range f.extra {
 		data[k] = v
 	}
-	if msg.PLogMetaData != nil {
+	if msg.PLogMetaData != nil && !msg.PLogMetaData.Last {
 		data["partial_message"] = "true"
```

Comment:
While we are here, lets also add the missing fields in the `data` map.

```data["partial_id"] = msg.PLogMetaData.ID
data["partial_ordinal"] = strconv.Itoa(msg.PLogMetaData.Ordinal)
data["partial_last"] = strconv.FormatBool(msg.PLogMetaData.Last)```

Reply:
This is already taken care of by https://github.com/moby/moby/pull/38065/files, but CI is broke on that. You can carry over this specific change from that PR.

Reply:
Yes, well, this approach is much better